wordpress-mobile/WordPress-Android
# Issue/10151 add tracks to post settings featured image actions

Attempt to address #10151

This PR adds analytics tracks to Post Settings - Featured Image functionality

To test:

- Create a draft or edit a published post
- Click on the overflow menu in the editor -> Post settings
- Click Set Featured Image, choose a photo, wait till image is successfully uploaded and set to image view
- Check that the console contains the following log:
    - `🔵 Tracked: post_settings_featured_image_picked, Properties: {"post_id":<post id>,"post_status":<post status>}`
- Tap on the featured image, click “Remove featured image”
- Check that the console contains the following log:
    - `🔵 Tracked: post_settings_featured_image_remove_clicked, Properties: {"post_id":<post id>,"post_status":<post status>}`
- Click Set Featured Image, choose a photo, tap on the local image while it is uploading and click “Remove featured image”
- Check that the console contains the following log:
    - `🔵 Tracked: post_settings_featured_image_upload_canceled, Properties: {"post_id":<post id>,"post_status":<post status>}`
- Click Set Featured Image, choose photo, disable network so that upload is cancelled, tap on photo to retry, click retry uploading 
- Check that the console contains the following log:
    - `🔵 Tracked: post_settings_featured_image_upload_retry_clicked, Properties: {"post_id":<post id>,"post_status":<post status>}`



PR submission checklist:
